### PR TITLE
Add types-s3transfer recipe

### PR DESCRIPTION
**Pull request by vemel**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/types-s3transfer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - python >=3.6
+
+outputs:
```
Comment (hmaarrfk) [inline recipes/types-s3transfer/meta.yaml:26]: is there a reason why you are using the split package syntax? It really complicates recipes

File: recipes/types-s3transfer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{% set name = "types-s3transfer" %}
+{% set version = "0.6.0.post1" %}
```
Comment (hmaarrfk) [inline recipes/types-s3transfer/meta.yaml:2]: do you need post1? it is somewhat discouraged to do pre-releases

File: recipes/types-s3transfer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      run:
+        - python >=3.6
+    build:
+      noarch: python
+      script:
+        - python -m pip install . -vv
+    test:
+      script: run_test.py
+      commands:
+        - pip check
```
Comment (hmaarrfk) [inline recipes/types-s3transfer/meta.yaml:38]:
```suggestion
        - pip check
        - python -c "import importlib; importlib.import_module('s3transfer-stubs')"
```

Is this really the right way to install stub files? I'm not too sure, but it seems strange to have package names with `dashes`.

File: recipes/types-s3transfer/meta.yaml (hunk 1 above)
Comment (vemel) [reply recipes/types-s3transfer/meta.yaml:26]: For some reason `test.script` is not accepted, but `ouputs.*.tests.script` works. Could please help me to rewrite it properly?

File: recipes/types-s3transfer/meta.yaml (hunk 2 above)
Comment (vemel) [reply recipes/types-s3transfer/meta.yaml:2]: The idea is that `types-s3transfer` version should match `s3transfer` version, so I use postreleases for hotfixes.

File: recipes/types-s3transfer/meta.yaml (hunk 3 above)
Comment (vemel) [reply recipes/types-s3transfer/meta.yaml:38]: Yes, stub package for `mylib` should be named `mylib-stubs`

File: recipes/types-s3transfer/meta.yaml (hunk 1 above)
Comment (vemel) [reply recipes/types-s3transfer/meta.yaml]: FIxed, thank you!

File: recipes/types-s3transfer/meta.yaml (hunk 3 above)
Comment (vemel) [reply recipes/types-s3transfer/meta.yaml]: Fixed, thanks for the suggestion.

File: recipes/types-s3transfer/meta.yaml (hunk 2 above)
Comment (hmaarrfk) [reply recipes/types-s3transfer/meta.yaml:2]: i hope the bot picks it up. i'm not sure it will.

**Review by hmaarrfk — APPROVED** [review]

**Pull request merged**